types/eth_client.go

```diff
@@ -49,3 +62,37 @@ type NoopRPCClient struct{}
 func (c *NoopRPCClient) Call(result any, method string, args ...any) error {
 	return nil
 }
+
+// DialWithRetry attempts to connect to an Ethereum client with retries and exponential backoff.
+// It returns an EthClienter on success or an error if all attempts fail.
+func DialWithRetry(ctx context.Context, url string,
+	maxRetries int, initialBackoff, maxBackoff time.Duration) (EthClienter, error) {
+	var (
+		client *ethclient.Client
+		err    error
+	)
+
+	// If maxRetries is 0, we try to connect once without retries.
+	if maxRetries == 0 {
+		maxRetries = 1
+	}
+
+	for attempt := range maxRetries {
```

Comment:
Maybe we can do something like this in the loop, it looks cleaner:

```go
client, err := ethclient.Dial(url)
if err == nil {
	return NewDefaultEthClient(client, client.Client()), nil
}

// On failure
backoff := float64(initialBackoff) * math.Pow(backoffBase, float64(attempt))
wait := time.Duration(math.Min(backoff, float64(maxBackoff)))

log.Warnf("Dialing %s failed (attempt %d/%d): %v. Retrying in %s...",
	url, attempt+1, maxRetries, err, wait)

// Wait or exit early if context is canceled
select {
case <-ctx.Done():
	return nil, ctx.Err()
case <-time.After(wait):
	// continue to next attempt
}
```

Reply:
Fixed https://github.com/agglayer/aggkit/pull/761/commits/bad3b5d7100e3d3d1c6c52ff64f246bc4d4b72a1